pkg/morph/deploy/notary.go

```diff
@@ -869,3 +878,353 @@ func makeUnsignedDesignateCommitteeNotaryTx(roleContract *rolemgmt.Contract, com
+	// cache processed operations: when main transaction from received notary
+	// request is signed and sent by the local account, we receive the request from
+	// the channel again
+	mProcessedMainTxs := make(map[util.Uint256]struct{})
+
+	subID, err := prm.blockchain.ReceiveNotaryRequests(&neorpc.TxFilter{
+		Signer: &committeeMultiSigAccID,
+	}, chNotaryRequests)
+	if err != nil {
+		return fmt.Errorf("subscribe to notary requests from committee: %w", err)
+	}
+
+	go func() {
+		defer func() {
+			err := prm.blockchain.Unsubscribe(subID)
+			if err != nil {
+				prm.logger.Warn("failed to cancel subscription to notary requests", zap.Error(err))
+			}
+		}()
+
+		prm.logger.Info("listening to committee notary requests...")
+
+		for {
+			select {
+			case <-ctx.Done():
+				prm.logger.Info("stop listening to committee notary requests (context is done)", zap.Error(ctx.Err()))
+				return
+			case notaryEvent, ok := <-chNotaryRequests:
+				if !ok {
+					prm.logger.Info("stop listening to committee notary requests (subscription channel closed)")
+					return
+				}
+
+				// for simplicity, requests are handled one-by one. We could process them in parallel
+				// using worker pool, but actions seem to be relatively lightweight
+
+				const expectedSigners = 3 // sender + committee + Notary
+				mainTx := notaryEvent.NotaryRequest.MainTransaction
+				// note: instruction above can throw NPE and it's ok to panic: we confidently
+				// expect that only non-nil pointers will come from the channel (NeoGo
+				// guarantees)
+
+				srcMainTxHash := mainTx.Hash()
+				_, processed := mProcessedMainTxs[srcMainTxHash]
+
+				switch {
+				case processed:
+					prm.logger.Info("main transaction of the notary request has already been processed, skip",
+						zap.Stringer("ID", srcMainTxHash))
+					continue
+				case notaryEvent.Type != mempoolevent.TransactionAdded:
+					prm.logger.Info("unsupported type of the notary request event, skip",
+						zap.Stringer("got", notaryEvent.Type), zap.Stringer("expect", mempoolevent.TransactionAdded))
+					continue
+				case len(mainTx.Signers) != expectedSigners:
+					prm.logger.Warn("unsupported number of signers of main transaction from the received notary request, skip",
+						zap.Int("expected", expectedSigners), zap.Int("got", len(mainTx.Signers)))
+					continue
+				case !mainTx.HasSigner(committeeMultiSigAccID):
+					// in theory, there can be any notary requests besides those sent by the current
+					// auto-deploy procedure. However, it's better to log with 'info' severity since
+					// this isn't really expected in practice.
+					prm.logger.Warn("committee is not a signer of main transaction from the received notary request, skip")
+					continue
+				case mainTx.HasSigner(prm.localAcc.ScriptHash()):
+					prm.logger.Info("main transaction from the received notary request is signed by a local account, skip")
+					continue
```

Comment:
It's the most significant question here, I'm not sure we have to create all these notary requests.

Reply:
> simultaneousely

in general, definitely not.

yep, we send different requests because we can't predict which actor will succeed, that's why i've implemented this logic as it is

> proxy+committee 

i didn't really understood how Proxy contract will help us. Who sends notary request to update the NNS contract in ur suggestion?

Reply:
Discussed, moved to a separate issue (need to add the link).

Reply:
#2416